### PR TITLE
Dev-python/spyder version bump 4.2.0

### PR DESCRIPTION
**Pull request by AndrewAmmerlaan**
bumps spyder and friends

### CODE REVIEW

File: dev-python/spyder/spyder-4.2.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,219 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{7,8} )
+
+inherit eutils xdg distutils-r1
```
Comment (juippis) [inline dev-python/spyder/spyder-4.2.0.ebuild]:
```
/var/tmp/portage/gpm/portage/dev-python/spyder-4.2.0/temp/environment: line 2528: optfeature: command not found
/var/tmp/portage/gpm/portage/dev-python/spyder-4.2.0/temp/environment: line 2529: optfeature: command not found
/var/tmp/portage/gpm/portage/dev-python/spyder-4.2.0/temp/environment: line 2530: optfeature: command not found
/var/tmp/portage/gpm/portage/dev-python/spyder-4.2.0/temp/environment: line 2531: optfeature: command not found
/var/tmp/portage/gpm/portage/dev-python/spyder-4.2.0/temp/environment: line 2532: optfeature: command not found
/var/tmp/portage/gpm/portage/dev-python/spyder-4.2.0/temp/environment: line 2533: optfeature: command not found
/var/tmp/portage/gpm/portage/dev-python/spyder-4.2.0/temp/environment: line 2534: optfeature: command not found
/var/tmp/portage/gpm/portage/dev-python/spyder-4.2.0/temp/environment: line 2535: optfeature: command not found
/var/tmp/portage/gpm/portage/dev-python/spyder-4.2.0/temp/environment: line 2536: optfeature: command not found
/var/tmp/portage/gpm/portage/dev-python/spyder-4.2.0/temp/environment: line 2537: optfeature: command not found
/var/tmp/portage/gpm/portage/dev-python/spyder-4.2.0/temp/environment: line 2538: optfeature: command not found
/var/tmp/portage/gpm/portage/dev-python/spyder-4.2.0/temp/environment: line 2539: optfeature: command not found
/var/tmp/portage/gpm/portage/dev-python/spyder-4.2.0/temp/environment: line 2540: optfeature: command not found
```

Comment (juippis) [inline dev-python/spyder/spyder-4.2.0.ebuild]:
```
 * DISTUTILS_USE_SETUPTOOLS value is probably incorrect
 *   have:     DISTUTILS_USE_SETUPTOOLS=bdepend (or unset)
 *   expected: DISTUTILS_USE_SETUPTOOLS=no
```

File: dev-python/spyder/spyder-4.2.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+
+BDEPEND="test? (
+	<dev-python/coverage-5.0[${PYTHON_USEDEP}]
+	dev-python/cython[${PYTHON_USEDEP}]
+	dev-python/flaky[${PYTHON_USEDEP}]
+	dev-python/matplotlib[tk,${PYTHON_USEDEP}]
+	dev-python/mock[${PYTHON_USEDEP}]
+	dev-python/pandas[${PYTHON_USEDEP}]
+	dev-python/pillow[${PYTHON_USEDEP}]
+	<dev-python/pytest-5.0[${PYTHON_USEDEP}]
```
Comment (juippis) [inline dev-python/spyder/spyder-4.2.0.ebuild]: Is this still the case?

**Review by juippis — APPROVED** [review]

File: dev-python/spyder/spyder-4.2.0.ebuild (hunk 2 above)
Comment (AndrewAmmerlaan) [reply dev-python/spyder/spyder-4.2.0.ebuild]: Yes

```
extras_require = {
    'test:platform_system == "Linux"': ['pytest-xvfb'],
    'test:platform_system == "Windows"': ['pywin32'],
    'test': [
        'coverage<5.0',
        'cython',
        'flaky',
        'matplotlib',
        'mock',
        'pandas',
        'pillow',
        'pytest<5.0',
        'pytest-cov',
        'pytest-faulthandler<2.0',
        'pytest-lazy-fixture',
        'pytest-mock',
        'pytest-ordering',
        'pytest-qt',
        'pyyaml',
        'scipy',
        'sympy',
    ],
}
```
https://github.com/spyder-ide/spyder/blob/master/setup.py

File: dev-python/spyder/spyder-4.2.0.ebuild (hunk 1 above)
Comment (AndrewAmmerlaan) [reply dev-python/spyder/spyder-4.2.0.ebuild]: Yes I saw that, but the warning is clearly wrong:

```
# Third party imports
from setuptools import setup, find_packages
from setuptools.command.install import 
```

https://github.com/spyder-ide/spyder/blob/master/setup.py

File: dev-python/pyls-spyder/pyls-spyder-0.1.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+EAPI=7
+
+PYTHON_COMPAT=( python3_{7,8} )
+
+inherit distutils-r1
+
+DESCRIPTION="Spyder extensions for the python language server"
+HOMEPAGE="https://github.com/spyder-ide/pyls-spyder
+	https://pypi.org/project/pyls-spyder/"
+SRC_URI="https://github.com/spyder-ide/${PN}/archive/v${PV}.tar.gz -> ${P}-gh.tar.gz"
```
[the remark this answers is not in the document]
Comment (AndrewAmmerlaan) [reply dev-python/pyls-spyder/pyls-spyder-0.1.1.ebuild:13]: Oops, seems like I copied this accidentality from another spyder related ebuild, will remove it next time I update this ebuild.

**Comment by AndrewAmmerlaan** [comment]
Thank you @juippis !

**Pull request closed without merging**